MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -526,161 +520,344 @@ public string GetAudioEncoder(EncodingJobInfo state)
             return codec.ToLowerInvariant();
         }
 
+        private string GetVideoToolboxDeviceArgs(string alias)
+        {
+            alias ??= VideotoolboxAlias;
+
+            // device selection in vt is not supported.
+            return " -init_hw_device videotoolbox=" + alias;
+        }
+
+        private string GetCudaDeviceArgs(int deviceIndex, string alias)
+        {
+            alias ??= CudaAlias;
+            deviceIndex = deviceIndex >= 0
```

Comment:
When you call `GetCudaDeviceArgs` it seems to always be 0. Is that on purpose?

Reply:
Yeah. We only support the first device0 for the time being. Multi device selection will be added after finishing the ‘ffcheck’ tool.